server/internal/infrastructure/memory/user.go

```diff
@@ -2,165 +2,135 @@ package memory
 
 import (
 	"context"
-	"sync"
 
 	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
 	"github.com/reearth/reearth-cms/server/pkg/id"
 	"github.com/reearth/reearth-cms/server/pkg/rerror"
 	"github.com/reearth/reearth-cms/server/pkg/user"
+	"github.com/reearth/reearth-cms/server/pkg/util"
 )
 
 type User struct {
-	lock sync.Mutex
-	data map[id.UserID]*user.User
+	data util.SyncMap[id.UserID, *user.User]
 }
 
 func NewUser() repo.User {
 	return &User{
-		data: map[id.UserID]*user.User{},
+		data: util.SyncMap[id.UserID, *user.User]{},
 	}
 }
 
 func (r *User) FindByIDs(ctx context.Context, ids id.UserIDList) ([]*user.User, error) {
-	r.lock.Lock()
-	defer r.lock.Unlock()
-
-	result := []*user.User{}
-	for _, id := range ids {
-		if d, ok := r.data[id]; ok {
-			result = append(result, d)
-		} else {
-			result = append(result, nil)
+	res := r.data.FindAll(func(key id.UserID, value *user.User) bool {
+		var res bool
+		for _, uid := range ids {
+			res = res || key == uid
```

Comment:
```suggestion
	res := r.data.FindAll(func(key id.UserID, value *user.User) bool {
		return ids.Has(key)
```

Confirm methods of `idx.List` and `idx.ID`. You can shorten many code lines